### PR TITLE
Adding Full VM Solution

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

Adding Full VM solution with a default 50 GB disk + verified Ubuntu flists.

### Changes

List of changes this PR includes:
- Creating a new Full VM solution and renaming the old one to be Micro VM

### Related Issues

- https://github.com/threefoldtech/grid_weblets/issues/759

### CODE REVIEW

File: src/utils/deployFullvm.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  return disk;
+}
+
+function createEnvs(envs: Env[]): { [key: string]: string } {
+  return envs.reduce((res, env) => {
+    res[env.key] = env.value;
+    return res;
+  }, {});
+}
+
```
Comment (rawdaGastan) [inline src/utils/deployFullvm.ts:68]: Why we didn't use deployVM instead of creating a new file?

File: src/types/fullvm.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import VM, { Disk } from "./vm";
+import { v4 } from "uuid";
+import validateName from '../utils/validateName';
+import isValidInteger from '../utils/isValidInteger';
+
+export default class FullVM extends VM {
+  public name = `VM${v4().split("-")[0]}`;
+  public diskSize: number = 50;
+  public disks = [new Disk(undefined, undefined, this.diskSize, undefined)];
```
Comment (rawdaGastan) [inline src/types/fullvm.ts:9]: I think the disk should be mounted on `/`

**Review by rawdaGastan — CHANGES_REQUESTED** [review]

File: src/elements/fullvm/Fullvm.wc.svelte [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+  import DeleteBtn from "../../components/DeleteBtn.svelte";
+  import AddBtn from "../../components/AddBtn.svelte";
+  import DeployBtn from "../../components/DeployBtn.svelte";
+  import Alert from "../../components/Alert.svelte";
+  import Modal from "../../components/DeploymentModal.svelte";
+  import hasEnoughBalance from "../../utils/hasEnoughBalance";
+  import validateName, {
+    isInvalid,
+    validateCpu,
+    validateDisk,
```
Comment (rawdaGastan) [inline src/elements/fullvm/Fullvm.wc.svelte:24]: Remove unused import

Comment (zaelgohary) [reply src/elements/fullvm/Fullvm.wc.svelte]: done.

File: src/types/fullvm.ts (hunk 2 above)
Comment (zaelgohary) [reply src/types/fullvm.ts]: done.

File: src/utils/deployFullvm.ts (hunk 1 above)
Comment (zaelgohary) [reply src/utils/deployFullvm.ts:68]: There were some requested changes as adding a default disk for the full VM and I preferred to do it in a separate file. Do you think it's better if I made it in the same file?

Comment (zaelgohary) [reply src/utils/deployFullvm.ts]: done.

File: src/elements/fullvm/Fullvm.wc.svelte [hunk 4]
```diff
@@ -287,7 +288,7 @@
           memory={data.memory}
           ssd={data.disks.reduce(
             (total, disk) => total + disk.size,
-            data.rootFs
+            data.diskSize
```
Comment (rawdaGastan) [inline src/elements/fullvm/Fullvm.wc.svelte:291]: why we add `data.diskSize` ?
I think it is already added.

**Review by rawdaGastan — CHANGES_REQUESTED** [review]

File: src/elements/fullvm/Fullvm.wc.svelte [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+    validateCpu,
+    validateEntryPoint,
+    validateFlistvalue,
+    validateKey,
+    validateKeyValue,
+    validateMemory,
+  } from "../../utils/validateName";
+  import { noActiveProfile } from "../../utils/message";
+  import isInvalidFlist from "../../utils/isInvalidFlist";
+  import RootFsSize from "../../components/RootFsSize.svelte";
```
Comment (rawdaGastan) [inline src/elements/fullvm/Fullvm.wc.svelte:32]: remove unused import 
`  import RootFsSize from "../../components/RootFsSize.svelte"`

Comment (zaelgohary) [reply src/elements/fullvm/Fullvm.wc.svelte]: done.

**Review by AhmedHanafy725 — APPROVED** [review]

**Review by rawdaGastan — APPROVED** [review]

**Pull request merged**